### PR TITLE
Enable edit customer even with a non existent coverage

### PR DESCRIPTION
**Pull request by pthegner**
# Description

This PR enable the edition of a customer who have an old coverage in order to change / delete it

## Issue

Issue link: BOT-52

## How to test

Here are the following steps to test this pull request:

- You must have a client set with a coverage who doesn't exist (to do this, you could change it in your database)
- Edit this client
- Delete, or change the coverage

### CODE REVIEW

File: Form/Type/PerimeterType.php [hunk 1]
```diff
@@ -69,21 +69,22 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 $externalCoverageId = (is_array($data) ? $data['external_coverage_id'] : $data->getExternalCoverageId());
                 $externalNetworkId = (is_array($data) ? $data['external_network_id'] : $data->getExternalNetworkId());
 
-                $form->remove('external_network_id');
-                $networks = $this->navitia->getNetWorks($externalCoverageId);
+                if (in_array($externalCoverageId, $this->coverages)) {
+                    $form->remove('external_network_id');
```
Comment (dvdn) [inline Form/Type/PerimeterType.php]: Hard choice to remove without information to user ?

Comment (pthegner) [inline Form/Type/PerimeterType.php]: remove ? it's just in the display. You could remove it afterwards

File: Form/Type/PerimeterType.php [hunk 2]
```diff
@@ -69,21 +69,22 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 $externalCoverageId = (is_array($data) ? $data['external_coverage_id'] : $data->getExternalCoverageId());
                 $externalNetworkId = (is_array($data) ? $data['external_network_id'] : $data->getExternalNetworkId());
 
-                $form->remove('external_network_id');
-                $networks = $this->navitia->getNetWorks($externalCoverageId);
+                if (in_array($externalCoverageId, $this->coverages)) {
```
Comment (datanel) [inline Form/Type/PerimeterType.php]: can you comment that please?
with your description for example "Enable edit customer even with a non existent coverage"

**Comment by dvdn** [comment]
2 awkward scenarii : 
A.
> GIVEN I have a bad coverage
> THEN I've got error Message OK
> GIVEN I valid the form
> THEN my coverage is removed without information about it
> THEN i've lost my coverage forever

B. (less important as everything remains persisted)
You have an outdated external ID (invalid).
You open the form, you've got the error message OK
BUT you can still save the form with this outdated networkID (but product OK still functionnal)

**Pull request merged**